packages/react/toolbar/src/Toolbar.tsx

```diff
@@ -0,0 +1,246 @@
+import * as React from 'react';
+import { RovingFocusGroup, useRovingFocus } from '@radix-ui/react-roving-focus';
+import { Primitive } from '@radix-ui/react-primitive';
+import { getSelector } from '@radix-ui/utils';
+import { createContext, composeEventHandlers } from '@radix-ui/react-utils';
+import type * as Polymorphic from '@radix-ui/react-polymorphic';
+import type { Merge } from '@radix-ui/utils';
+
+/* -------------------------------------------------------------------------------------------------
+ * Toolbar
+ * -----------------------------------------------------------------------------------------------*/
+
+const TOOLBAR_NAME = 'Toolbar';
+const TOOLBAR_CONTEXT_NAME = TOOLBAR_NAME + 'Context';
+const TOOLBAR_DEFAULT_TAG = 'div';
+
+type Orientation = React.AriaAttributes['aria-orientation'];
+type Direction = 'ltr' | 'rtl';
+
+export type RovingFocusGroupOptions = {
+  orientation?: Orientation;
+  dir?: Direction;
+  loop?: boolean;
+};
+
+type ToolbarOwnProps = Merge<
+  Polymorphic.OwnProps<typeof Primitive>,
+  {
+    /**
+     * The orientation of the toolbar.
+     * Mainly so arrow navigation is done accordingly (left & right vs. up & down)
+     * (default: horizontal)
+     */
+    orientation?: React.AriaAttributes['aria-orientation'];
+    /**
+     * The direction of navigation between toolbar items.
+     * (default: ltr)
+     */
+    dir?: Direction;
+    /**
+     * Whether keyboard navigation should loop focus
+     * (default: true)
+     */
+    loop?: boolean;
+  }
+>;
+
+type ToolbarPrimitive = Polymorphic.ForwardRefComponent<
+  typeof TOOLBAR_DEFAULT_TAG,
+  ToolbarOwnProps
+>;
+
+type ToolbarContextValue = {
+  name: typeof TOOLBAR_CONTEXT_NAME;
+  orientation: Orientation;
+};
+
+const [ToolbarContext, useToolbarContext] = createContext<ToolbarContextValue>(
+  TOOLBAR_CONTEXT_NAME,
+  TOOLBAR_NAME
+);
+
+const Toolbar = React.forwardRef((props, forwardedRef) => {
+  const {
+    children,
+
+    'aria-label': ariaLabel,
+    'aria-labelledby': ariaLabelledBy,
+    'aria-controls': ariaControls,
+
+    orientation = 'horizontal',
+    dir,
+    loop,
```

Comment:
TBF I don't think we've considered that everywhere, I wouldn't be surprised if I've made similar assumptions elsewhere on other abstractions.

Reply:
oh really? oops... 😅  I think there are slight differences (but not sure where yours falls without looking). 

So, if we did this:

```jsx
type ToolbarOwnProps = Polymorphic.OwnProps<typeof RovingFocusGroup>;
```

Then, it would be fine because it would inherit all of the logic **and tsdocs** from `RovingFocusGroup`. So if the tsdocs and code changed in `RovingFocusGroup` then `Toolbar` would still communicate all the right information. The difference here is that we are redefining tsdoc in `Toolbar` but the tsdoc could get out of sync with the code if `RovingFocusGroup` changed. 

Also, there's another layer to consider. Do we think `Toolbar` should always `loop` by default or that it doesn't really matter what it does by default? If the answer is the former, then it should really have code itself to ensure that by providing a default, otherwise we fall into the latter category.

Reply:
Yeah I definitely see what you're saying. I'm thinking we've probably made some similar assumptions around default values with components that use `FocusScope` and `DismissableLayer` for example. Maybe not all props, but I'm thinking perhaps some…

Reply:
Maybe, one for the todo list to audit 😅